File: frontend/src/app/graph/RankingRequest.ts
export class RankingRequest {
  nodesNumber: number;
  nodeMinWeight: number;
  nodeMaxWeight: number;
  linkMinWeight: number;
  linkMaxWeight: number;

  method: {
    name?: string;
    params: {
      limit?: number;
      iterations?: number;
    }
  };

  constructor(nodesNumber: number, nodeMinWeight: number, nodeMaxWeight: number, linkMinWeight: number, linkMaxWeight: number, method: { name?: string; params: { limit?: number; iterations?: number } }) {
    this.nodesNumber = nodesNumber;
    this.nodeMinWeight = nodeMinWeight;
    this.nodeMaxWeight = nodeMaxWeight;
    this.linkMinWeight = linkMinWeight;
    this.linkMaxWeight = linkMaxWeight;
    this.method = method;
  }
}

File: backend/src/main/java/com/graphs/entity/RankingRequest.java
package com.graphs.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RankingRequest {
    private int nodesNumber;
    private double nodeMinWeight;
    private double nodeMaxWeight;
    private double linkMinWeight;
    private double linkMaxWeight;

    private Method method;
}
